From 56fd571a48f4bb7eb45a7b859b318411b3313895 Mon Sep 17 00:00:00 2001
From: VinceChang6637 <vince_chang@aspeedtech.com>
Date: Fri, 15 Mar 2024 13:50:42 +0800
Subject: [PATCH] ast2700-dcscm: add dbus for x86-power-control

This ia a workaround solution.
AST2700 A0 doesn't support SGPIO Slave interrupt.
The power-status-sync.sh will update SGPIO value to DBus.

Signed-off-by: VinceChang6637 <vince_chang@aspeedtech.com>
---
 .../openbmc_project/State/Host.interface.yaml | 21 +++++++++++++++++++
 1 file changed, 21 insertions(+)

diff --git a/yaml/xyz/openbmc_project/State/Host.interface.yaml b/yaml/xyz/openbmc_project/State/Host.interface.yaml
index 006a501..ac72d20 100644
--- a/yaml/xyz/openbmc_project/State/Host.interface.yaml
+++ b/yaml/xyz/openbmc_project/State/Host.interface.yaml
@@ -25,6 +25,27 @@ properties:
       description: >
           A property to define restart cause of a host.
 
+    - name: PostComplete
+      type: boolean
+      default: false
+      description: >
+          This is for AST2700 DCSCM A0 SGPIOS workaround.
+          PostComplete property for x86-power-control
+
+    - name: PowerButton
+      type: boolean
+      default: false
+      description: >
+          This is for AST2700 DCSCM A0 SGPIOS workaround.
+          PowerButton property for x86-power-control
+
+    - name: PowerOk
+      type: boolean
+      default: false
+      description: >
+          This is for AST2700 DCSCM A0 SGPIOS workaround.
+          PowerOk property for x86-power-control
+
 enumerations:
     - name: Transition
       description: >
